pymc_marketing/product_incrementality/mv_its.py

```diff
@@ -13,176 +13,217 @@
-        treatment_time,
         existing_sales: list[str],
-        treatment_sales: str,
         market_saturated: bool = True,
-        rng=42,
-        sample_kwargs: dict | None = None,
+        model_config: dict | None = None,
+        sampler_config: dict | None = None,
     ):
-        self.data = data
-        self.treatment_time = treatment_time
+        super().__init__(model_config=model_config, sampler_config=sampler_config)
+
         self.existing_sales = existing_sales
-        self.treatment_sales = treatment_sales
-        self.rng = rng
-        self.sample_kwargs = sample_kwargs if sample_kwargs is not None else {}
         self.market_saturated = market_saturated
 
-        self.model = self.build_model(
-            self.data[self.existing_sales],
-            self.data[self.treatment_sales],
-            self.market_saturated,
-            treatment_time=self.treatment_time,
-        )
-        self.sample_prior_predictive()
-        self.fit()
-        self.sample_posterior_predictive()
-        self.calculate_counterfactual()
-        return
+    def create_idata_attrs(self) -> dict[str, str]:
+        """Create the attributes for the InferenceData object."""
+        attrs = super().create_idata_attrs()
+        attrs["existing_sales"] = json.dumps(self.existing_sales)
+        attrs["market_saturated"] = json.dumps(self.market_saturated)
 
-    @staticmethod
-    def build_model(
-        existing_sales: pd.DataFrame,
-        treatment_sales: pd.Series,
-        market_saturated: bool,
-        treatment_time,
-        *,
-        alpha_background=0.5,
-    ):
-        """Return a PyMC model for a multivariate interrupted time series analysis."""
-        if not existing_sales.index.equals(treatment_sales.index):
-            raise ValueError(  # pragma: no cover
-                "Index of existing_sales and treatment_sales must match."
-            )
+        return attrs
+
+    @classmethod
+    def attrs_to_init_kwargs(cls, attrs) -> dict[str, Any]:
+        """Convert the attributes of the InferenceData object to the __init__ kwargs."""
+        return {
+            "existing_sales": json.loads(attrs["existing_sales"]),
+            "market_saturated": json.loads(attrs["market_saturated"]),
+        }
 
-        # note: type hints for coords required for mypi to not get confused
-        coords: dict[str, list[str]] = {
-            "background_product": list(existing_sales.columns),
-            "time": list(existing_sales.index.values),
+    @property
+    def default_model_config(self) -> dict:
+        """Default model configuration."""
+        return {
+            "intercept": Prior("Normal", dims="background_product"),
+            "likelihood": Prior(
+                "TruncatedNormal",
+                lower=0,
+                sigma=Prior("HalfNormal", dims="background_product"),
+                dims=("time", "background_product"),
+            ),
+            "alpha_background": 0.5,
```

Comment:
what is the meaning of this `alpha_background` :D ?

Reply:
@drbenvincent

Reply:
I have moved this entirely to the `model_config` which will have a default like before but it also exposes it to the user to create (with some limitations)